consensus/spos/bls/sovereignSubRoundSignatureOutGoingTxData.go

```diff
@@ -0,0 +1,69 @@
+package bls
+
+import (
+	"fmt"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-go/consensus"
+	"github.com/multiversx/mx-chain-go/consensus/spos"
+	"github.com/multiversx/mx-chain-go/errors"
+)
+
+type sovereignSubRoundSignatureOutGoingTxData struct {
+	signingHandler consensus.SigningHandler
+}
+
+// NewSovereignSubRoundSignatureOutGoingTxData creates a new signer for sovereign outgoing tx data in signature sub round
+func NewSovereignSubRoundSignatureOutGoingTxData(signingHandler consensus.SigningHandler) (*sovereignSubRoundSignatureOutGoingTxData, error) {
+	if check.IfNil(signingHandler) {
+		return nil, spos.ErrNilSigningHandler
+	}
+
+	return &sovereignSubRoundSignatureOutGoingTxData{
+		signingHandler: signingHandler,
+	}, nil
+}
+
+// CreateSignatureShare creates a signature share for outgoing tx hash, if exists
+func (sr *sovereignSubRoundSignatureOutGoingTxData) CreateSignatureShare(
+	header data.HeaderHandler,
+	selfIndex uint16,
+	selfPubKey []byte,
+) ([]byte, error) {
+	sovChainHeader, castOk := header.(data.SovereignChainHeaderHandler)
+	if !castOk {
+		return nil, fmt.Errorf("%w in sovereignSubRoundSignatureOutGoingTxData.CreateSignatureShare", errors.ErrWrongTypeAssertion)
+	}
+
+	outGoingMBHeader := sovChainHeader.GetOutGoingMiniBlockHeaderHandler()
+	if check.IfNil(outGoingMBHeader) {
+		return make([]byte, 0), nil
+	}
+
+	return sr.signingHandler.CreateSignatureShareForPublicKey(
```

Comment:
this signs the `outGoingMBHeader.GetOutGoingOperationsHash()` 
what will be actually put in the txData for the outgoing transaction? the verification should be very easy to compute on the bridgeSC. Will the bridgeSC have access to all txsin the outGoindMBHeader to create the MBHeader and check the signature? ?

Reply:
it will check that data is multiSigned by the  validators.

Reply:
Yes, that's the hash of the tx data outgoing tx;
SC Bridge will need to be provided all info